code/edgemedia/src/androidTest/java/com/adobe/marketing/mobile/util/FunctionalTestHelper.java

```diff
@@ -0,0 +1,952 @@
+     * @param responseMessage the network response message, returned by {@link
+     *     HttpConnecting#getResponseMessage()}
+     * @param propertyMap the network response header map, returned by {@link
+     *     HttpConnecting#getResponsePropertyValue(String)}
+     * @return an {@link HttpConnecting} object
+     * @see #setNetworkResponseFor(String, HttpMethod, HttpConnecting)
+     */
+    public static HttpConnecting createNetworkResponse(
+            final String responseString,
+            final String errorString,
+            final int code,
+            final String responseMessage,
+            final Map<String, String> propertyMap) {
+        return new HttpConnecting() {
+            @Override
+            public InputStream getInputStream() {
+                if (responseString != null) {
+                    return new ByteArrayInputStream(
+                            responseString.getBytes(StandardCharsets.UTF_8));
+                }
+
+                return null;
+            }
+
+            @Override
+            public InputStream getErrorStream() {
+                if (errorString != null) {
+                    return new ByteArrayInputStream(errorString.getBytes(StandardCharsets.UTF_8));
+                }
+
+                return null;
+            }
+
+            @Override
+            public int getResponseCode() {
+                return code;
+            }
+
+            @Override
+            public String getResponseMessage() {
+                return responseMessage;
+            }
+
+            @Override
+            public String getResponsePropertyValue(String responsePropertyKey) {
+                if (propertyMap != null) {
+                    return propertyMap.get(responsePropertyKey);
+                }
+
+                return null;
+            }
+
+            @Override
+            public void close() {}
+        };
+    }
+
+    /**
+     * Sets the provided delay for all network responses, until reset
+     *
+     * @param delaySec delay in seconds
+     */
+    public static void enableNetworkResponseDelay(final Integer delaySec) {
+        if (delaySec < 0) {
+            return;
+        }
+
+        testNetworkService.enableDelayedResponse(delaySec);
+    }
+
+    /**
+     * Pause test execution for the given {@code milliseconds}
+     *
+     * @param milliseconds the time to sleep the current thread.
+     */
+    public static void sleep(int milliseconds) {
+        try {
+            Thread.sleep(milliseconds);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Use this API for JSON formatted {@code NetworkRequest} body in order to retrieve a flattened
+     * map containing its data.
+     *
+     * @param networkRequest the {@link NetworkRequest} to parse
+     * @return The JSON request body represented as a flatten map
+     */
+    public static Map<String, String> getFlattenedNetworkRequestBody(
```

Comment:
It doesn't look like you are currently using this function to get a "flattened" network request body. In your tests, you are getting the request body as a Map<String, Any> and doing the comparison that way, which works better in this case. However, this method is the only place where I see the FunctionalTestUtils being used. Unless you think we will need this function in the future, I'd like to remove it along with the entire FunctionalTestUtils class to remove the dead code.